zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -285,6 +298,20 @@ protected Object jobRun() throws Throwable {
       throw new RuntimeException("Can not find interpreter for " + getRequiredReplName());
     }
 
+    if (this.user != null &&
+      !factory.getInterpreterSettings(note.getId()).isEmpty()) {
+      for (InterpreterSetting intp: factory.getInterpreterSettings(note.getId())){
+
+        if (replName.startsWith(intp.getName()) &&
+          intp.getOption().isSetPermission() &&
+          !hasPermission(authenticationInfo.getUser(), intp.getOption().getUsers())) {
+          logger.error("{} has no permission for {} ", authenticationInfo.getUser(), repl);
```

Comment:
All your comments are totally makes sense and thank you very much for the detail review to spend your precious time.
I'll fix them. 
Thank you again.